### ISSUE TITLE
editors, lines and cols

### ISSUE THREAD

**Issue by vixns** [criteo/mesos-term#12]
When launching an editor in mesos-term

- nano returns an error `Error opening terminal: unknown.`, we have to manually set the TERM env  `TERM=linux nano`
- vim, nano, and I guess most editors use 24 lines and 80 cols, regardless the window size.

2 questions : 
1 / shouldn't TERM=linux be injected in the environment if not present ?
2 / how can we use the full window with text editors ?

**Comment by vixns** [comment]
The latest xterm.js release may help solving this issue
https://github.com/xtermjs/xterm.js/releases/tag/3.11.0

**Comment by clems4ever** [comment]
Hello @vixns , this problem is resolved with the following PR that has been merged these days: https://github.com/clems4ever/mesos-term/pull/17.

**Comment by clems4ever** [comment]
Regarding the resizing problem, we need to try the upgrade of xtermjs on our side.

**Comment by vixns** [comment]
After more testing, I found it only occurs when running mesos-term within a container.

It seems related to signal propagation, `/resize...` ajax calls are ignored, the `SIGWINCH`  signal is fired but the python program does not receive it when running within docker, so `stty` is not called.

On my development laptop within docker `stty` is called only once on load, on each resize without docker.
Running in a mesos container, resize is never called (race condition on the first call? this is probably a separate issue).

Workaround :
Open the browser's inspector's network tab, resize the window, you'll see the cols and rows values in `/resize...` calls. Send the command `stty cols XXX rows XXX` using these values.

**Comment by vixns** [comment]
The culprit was alpine, rebasing the image on node:lts-slim, which is debian, fix the signal propagation issue.

**Comment by clems4ever** [comment]
hello @vixns , I'm rewriting MesosTerm in React to integrate sandbox browsing to the application. I tested resizing and it seems to work in the new version: https://github.com/clems4ever/mesos-term/pull/28.

**Comment by kamaradclimber** [comment]
Closing this issue, feel free to reopen if the issue persist.
